### ISSUE TITLE
[Analyzer] False positive uninitialised member when brace initialization used

### ISSUE THREAD

**Issue by dd8** [llvm/llvm-project#98128]
The following code produces an analyzer warning that member_pointer is garbage or undefined:

```
class One
{
public:
	char* member_pointer = nullptr;
};

class Two
{
public:
	typedef enum
	{
		nameNone = 1,
		nameTwo = 2,
		
	} Name;
	const Name member_name = nameNone;
	One member_one;
};

char* Test()
{
	Two* two = new Two{ Two::nameTwo };
	auto pointer = two->member_one.member_pointer; // reports assigned value is garbage or undefined
	return pointer;
}
```

This looks incorrect because default initialisation is done on any members not initialised by the brace initializer:
https://en.cppreference.com/w/cpp/language/aggregate_initialization#Implicitly_initialized_elements

I've confirmed member_pointer in the code above is initialised to zero when testing with godbolt.org using GCC/CLang/MSVC

**Comment by dd8** [comment]
clang-tidy 18.1.0 on Godbolt reports the same thing, as does clang trunk:

x86-64 clang 18.1.0 - 962ms (18055B) ~340 lines filtered
clang-tidy (trunk) x86-64 clang 18.1.0 (Editor #1, Compiler #1)
[<source>:23:2: warning: Assigned value is garbage or undefined [clang-analyzer-core.uninitialized.Assign]](javascript:;)
   23 |         auto pointer = two->member_one.member_pointer; // reports assigned value is garbage or undefined
      |         ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[<source>:22:13: note: Uninitialized value stored to field 'member_pointer'](javascript:;)
   22 |         Two* two = new Two{ Two::nameTwo };
      |                    ^~~~~~~~~~~~~~~~~~~~~~~
[<source>:23:2: note: Assigned value is garbage or undefined](javascript:;)
   23 |         auto pointer = two->member_one.member_pointer; // reports assigned value is garbage or undefined
      |         ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 warning generated.

**Comment by chrchr-github** [comment]
Reduced: 
~~~c++
struct S {
	char* p;
};

struct T {
	S s;
};

char* Test() {
	T* t = new T{};
	auto pointer = t->s.p; // reports assigned value is garbage or undefined
	return pointer;
}
~~~
https://godbolt.org/z/cTYT98dz4

**Comment by chrchr-github** [comment]
Even further reduced:
~~~c++
struct T {
    int a, b;
};

int main() {
	T* t = new T{};
	return t->b;
}
~~~
https://godbolt.org/z/dM9ofKxfv
`return t->a;` is clean.
Maybe related: https://github.com/llvm/llvm-project/issues/78420
